web/main/models.py

```diff
@@ -2021,12 +2024,33 @@ class ContentNode(
     resource_id = models.BigIntegerField(blank=True, null=True)
 
     objects = _ContentNodeManager()
+
     tracked_fields = ["headnote"]
 
     # Stores the number of ‘read’ characters in a content
     # Length of content, less elided text and html tags.
     reading_length = models.IntegerField(null=True)
 
+    is_instructional_material = models.BooleanField(
+        default=False,
+        help_text="This content should only be made available on the frontend to verified professors, staff, or users with editing privilege",
+    )
+
+    @classmethod
+    def nodes_for_user_by_casebook(
+        cls, casebook: Casebook, user: Optional[Union[User, AnonymousUser]], **kwargs
+    ) -> ContentNodeQuerySet:
+        if user:
+            if user.is_authenticated and (
+                user.verified_professor
+                or casebook.contentcollaborator_set.filter(user=user).exists()
+            ):
+                return ContentNode.objects.filter(casebook=casebook, **kwargs)
+            return ContentNode.objects.filter(casebook=casebook, **kwargs).exclude(
+                is_instructional_material=True
+            )
+        return ContentNode.objects.filter(casebook=casebook, **kwargs)
```

Comment:
I'm a little confused by the signature of this method. (I know the type is propagating down from higher up the call chain; commenting here for simplicity.) With `user: Optional[Union[User, AnonymousUser]]`, I would have expected that meant there would be a default `user=None`, so that we could call the method without specifying a user. In what situations do we anticipate calling it intentionally with `None`?

Also just double checking: if we *do* call it intentionally with `None`, should the fallback be to return all the nodes, as though an admin/qualifying user were passed in (as written), or should the fallback be more restrictive, showing only public nodes?

Just making sure this method wants to return all the nodes when `user` is `None`. If we forget to pass it a `user`,

Reply:
It's confusing because it's wrong!

Earlier in the code it was `user: Optional[User] = None` with a different method name. Removed `Optional` in both cases where it appeared.

Reply:
Aha!

I we can lose the `if user` check, then, too? Maybe in a followup PR.